androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/AppNavigation.kt

```diff
@@ -28,48 +28,50 @@ sealed class PatientMeetingScreen(val route: String) {
 
     object Landing : PatientMeetingScreen("{visitId}") {
         // Overloads and creates a route with a specific id
-        fun createRoute(visitId: String) = "$root/$visitId"
+        override fun createRoute(visitId: String) = "$root/$visitId"
     }
 
     object Arrival : PatientMeetingScreen("{visitId}/arrival") {
-        fun createRoute(visitId: String) = "$root/$visitId/arrival"
+        override fun createRoute(visitId: String) = "$root/$visitId/arrival"
     }
 
     object HazardAssessment : PatientMeetingScreen("{visitId}/hazardAssessment") {
-        fun createRoute(visitId: String) = "$root/$visitId/hazardAssessment"
+        override fun createRoute(visitId: String) = "$root/$visitId/hazardAssessment"
     }
 
     object ContactReason : PatientMeetingScreen("{visitId}/contactReason") {
-        fun createRoute(visitId: String) = "$root/$visitId/contactReason"
+        override fun createRoute(visitId: String) = "$root/$visitId/contactReason"
     }
 
     object PreviousCare : PatientMeetingScreen("{visitId}/previousCare") {
-        fun createRoute(visitId: String) = "$root/$visitId/previousCare"
+        override fun createRoute(visitId: String) = "$root/$visitId/previousCare"
     }
 
     object HealthHistory : PatientMeetingScreen("{visitId}/healthHistory") {
-        fun createRoute(visitId: String) = "$root/$visitId/healthHistory"
+        override fun createRoute(visitId: String) = "$root/$visitId/healthHistory"
     }
 
     object HealthNow : PatientMeetingScreen("{visitId}/healthNow") {
-        fun createRoute(visitId: String) = "$root/$visitId/healthNow"
+        override fun createRoute(visitId: String) = "$root/$visitId/healthNow"
     }
 
     object SuicideAssessment : PatientMeetingScreen("{visitId}/suicideAssessment") {
-        fun createRoute(visitId: String) = "$root/$visitId/suicideAssessment"
+        override fun createRoute(visitId: String) = "$root/$visitId/suicideAssessment"
     }
 
     object NursingNeed : PatientMeetingScreen("{visitId}/nursingNeed") {
-        fun createRoute(visitId: String) = "$root/$visitId/nursingNeed"
+        override fun createRoute(visitId: String) = "$root/$visitId/nursingNeed"
     }
 
     object MedicalOrder : PatientMeetingScreen("{visitId}/medicalOrder") {
-        fun createRoute(visitId: String) = "$root/$visitId/medicalOrder"
+        override fun createRoute(visitId: String) = "$root/$visitId/medicalOrder"
     }
 
     object InterimJournal : PatientMeetingScreen("{visitId}/interimJournal") {
-        fun createRoute(visitId: String) = "$root/$visitId/interimJournal"
+        override fun createRoute(visitId: String) = "$root/$visitId/interimJournal"
     }
+
+    abstract fun createRoute(visitId: String): String
```

Comment:
I prefer to have abstract methods, frequently used variables etc. at the top (like how some scripting languages require symbols to be declared before use) as I find it makes the class easier to understand at a glance.